### PR TITLE
add bbcode to tinyMCE

### PR DESCRIPTION
**Pull request by itkg-rboessel**

### CODE REVIEW

**Comment by itkg-ngilain** [comment]
:+1:

File: BackofficeBundle/Resources/public/coffee/orchestraLib.coffee [hunk 1]
```diff
@@ -159,7 +159,115 @@ callback_tinymce_init = null
 
 activateTinyMce = (view, textarea) ->
   do (view, textarea) ->
+    tinymce.html2bbcodeOrchestra =
```
Comment (nicolasThal) [inline BackofficeBundle/Resources/public/coffee/orchestraLib.coffee]: could you extract the activateTinyMce in another file

File: BackofficeBundle/Resources/config/form.yml [hunk 2]
```diff
@@ -44,9 +44,9 @@ parameters:
                     - advlist autolink lists link image charmap print preview hr anchor pagebreak
                     - searchreplace wordcount visualblocks visualchars code fullscreen
                     - insertdatetime media nonbreaking save table contextmenu directionality
-                    - emoticons template paste textcolor
+                    - emoticons template paste textcolor orchestra
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/config/form.yml]: may be rename the plugin name by something more explicit like orchestrabbcode

File: BackofficeBundle/Resources/public/coffee/orchestraLib.coffee [hunk 3]
```diff
@@ -159,7 +159,115 @@ callback_tinymce_init = null
...
+      '\\[\\/ul\\]'                    : '</ul>'              ,
+      '\\[ul\\]'                       : '<ul>'               ,
+      '\\[ul="(.*?)"\\]'               : '<ul style="$1">'    ,
+      '\\[span="(.*?)"\\]'             : '<span style="$1">'  ,
+      '\\[\\/span\\]'                  : '</span>'            ,
+      '\\[\\/li\\]'                    : '</li>'              ,
+      '\\[li\\]'                       : '<li>'
+
+    initParameter()
+    tinymce.create 'tinymce.plugins.BBCodeOrchestraPlugin',
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/public/coffee/orchestraLib.coffee]: can you extract the plugin in other file

File: BackofficeBundle/Resources/public/coffee/orchestraLib.coffee [hunk 4]
```diff
@@ -159,7 +159,115 @@ callback_tinymce_init = null
 
 activateTinyMce = (view, textarea) ->
   do (view, textarea) ->
+    tinymce.html2bbcodeOrchestra =
+      '<img.*?src="(.*?)".*?alt="(.*?)".*?title="(.*?)"\\/>' : '[img alt="$2" title="$3"]$1[/img]',
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/public/coffee/orchestraLib.coffee]: i  think this bbcode tag should be adding in mediaAdmin bundle

Comment (nicolasThal) [inline BackofficeBundle/Resources/public/coffee/orchestraLib.coffee]: Nice idea, each bundle should be able to add it's own tag to the main configuration

File: BackofficeBundle/Resources/public/coffee/orchestraLib.coffee [hunk 5]
```diff
@@ -159,7 +159,115 @@ callback_tinymce_init = null
...
+      '\\[i\\]'                        : '<em>'               ,
+      '\\[\\/i\\]'                     : '</em>'              ,
+      '\\[\\b\\]'                      : '<b>'                ,
+      '\\[\\/b\\]'                     : '</b>'               ,
+      '\\[\\/p\\]'                     : '</p>'               ,
+      '\\[p\\]'                        : '<p>'                ,
+      '\\[p="(.*?)"\\]'                : '<p style="$1">'     ,
+      '\\[br\\]'                       : '<br/>'              ,
+      '\\[\\/h1\\]'                    : '</h1>'              ,
+      '\\[h1\\]'                       : '<h1>'               ,
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/public/coffee/orchestraLib.coffee]: you can make a regex which match all title ?

**Comment by itkg-rboessel** [comment]
@itkg-alavieille @nicolasThal done

File: BackofficeBundle/Resources/public/coffee/tinyMCE/bbcode2htmlConfigurator.coffee [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    '\\[td\\]'                       : '<td>'               ,
+    '\\[td="([^"]*)"\\]'             : '<td style="$1">'    ,
+    '\\[\\/th\\]'                    : '</th>'              ,
+    '\\[th\\]'                       : '<th>'               ,
+    '\\[th="([^"]*)"\\]'             : '<th style="$1">'    ,
+    '\\[url="([^";]*)"](.*?)\\[\\/url\\]' : '<a href="$1">$2</a>',
+    '\\[td style="([^"]*)" scope="([^"]*)"\\]' : '<td style="$1" scope="$2">',
+    '\\[tr style="([^"]*)" scope="([^"]*)"\\]' : '<tr style="$1" scope="$2">',
+    '\\[th style="([^"]*)" scope="([^"]*)"\\]' : '<th style="$1" scope="$2">',
+    '\\[img alt="([^"]*)" width="([^"]*)" height="([^"]*)"\\]([^"]*)\\[\\/img\\]' : '<img src="$4" alt="$1" width="$2" height="$3"/>',
```
Comment (nicolasThal) [inline BackofficeBundle/Resources/public/coffee/tinyMCE/bbcode2htmlConfigurator.coffee]: all the transformations with the images should come from the media-admin-bundle

File: BackofficeBundle/Resources/public/coffee/tinyMCE/html2bbcodeConfigurator.coffee [hunk 7]
```diff
@@ -0,0 +1,96 @@
+html2bbcodeConfigurator = ->
+  transformation:
+    '<img.*?src="([^"]*)" alt="([^"]*)" width="([0-9]*)" height="([0-9]*)"\\/>' : '[img alt="$2" width="$3" height="$4"]$1[/img]',
```
Comment (nicolasThal) [inline BackofficeBundle/Resources/public/coffee/tinyMCE/html2bbcodeConfigurator.coffee]: same

**Comment by itkg-rboessel** [comment]
@nicolasThal @itkg-alavieille

**Comment by nicolasThal** [comment]
:+1:

**Comment by itkg-rboessel** [comment]
@nicolasThal I change some line, Can you check again ?

**Comment by itkg-rboessel** [comment]
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/24

**Comment by nicolasThal** [comment]
:+1:

**Pull request merged**